### PR TITLE
oxcfxics: latest messages first when syncing

### PR DESCRIPTION
**Pull request by jkerihuel**
Sort by PidTagMessageDeliveryTime descending order. 
Most recent delivery time first when available.

### CODE REVIEW

File: mapiproxy/servers/default/emsmdb/oxcfxics.c [hunk 1]
```diff
@@ -856,6 +857,19 @@ static bool oxcfxics_push_messageChange(struct emsmdbp_context *emsmdbp_ctx, str
 			DEBUG(5, ("push_messageChange: %d objects in table\n", table_object->object.table->denominator));
 			/* fetch maching mids */
 			message_sync_data->mids = talloc_array(message_sync_data, uint64_t, table_object->object.table->denominator);
+
+			/* sort messages: most recent delivery time first when available */
+			lpSortCriteria.cSorts = 1;
+			lpSortCriteria.cCategories = 0;
+			lpSortCriteria.cExpanded = 0;
+			lpSortCriteria.aSort = talloc_array(mem_ctx, struct SSortOrder, 1);
```
Comment (sixstone-qq) [inline mapiproxy/servers/default/emsmdb/oxcfxics.c]: I miss OOM check.

Comment (jkerihuel) [inline mapiproxy/servers/default/emsmdb/oxcfxics.c]: correct

**Comment by sixstone-qq** [comment]
Following git commit message guidelines would be ideal:

http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html

:)

File: mapiproxy/servers/default/emsmdb/oxcfxics.c (hunk 1 above)
Comment (sixstone-qq) [inline mapiproxy/servers/default/emsmdb/oxcfxics.c]: fair enough

**Comment by blaxter** [comment]
:+1: though commit message could be improved. I think somethig like:

    oxcfxics: latest messages first when syncing

    Sort by PidTagMessageDeliveryTime desc. Most recent 
    delivery time first when available.

would be so much better for future readers of git log.

**Comment by jkerihuel** [comment]
ack'd on my way to change commit msg

**Comment by blaxter** [comment]
mmm I think you tried to rebase the branch into the git ci --amend local branch?.

Easy way to reset commit message:

    git checkout  jkerihuel/newmsg-first
    git reset --hard 9adba15168aebc418c55fa6da4045107c938dc9a
    git commit --amend # and edit the message as you want it
    git push --force jkerihuel/newmsg-first:jkerihuel/newmsg-first

**Comment by jkerihuel** [comment]
good catch, let me redo

**Comment by blaxter** [comment]
Travis network is failing a lot the last days :disappointed: We had green after `ci --amend` so merging it

**Pull request merged**